modules/imgproc/src/min_enclosing_convex_polygon.cpp

```
@@ -0,0 +1,1185 @@
+ * @param k              Number of vertices of the output polygon
+ * @param minPolygon     Minimum area convex k-gon enclosing the given polygon
+ * @param area           Area of minPolygon
+ */
+static void findMinEnclosingPolygon(const std::vector<cv::Point2f> &ngon,
+                                    const int &k,
+                                    cv::OutputArray &minPolygon,
+                                    double &area) {
+    try
+    {
+        if (ngon.size() < 3)
+        {
+            throw std::invalid_argument(
+            "ngon must have 3 or more different points enclosing an area" );
+        }
+        if (cv::contourArea(ngon) < EPSILON )
+        {
+            throw std::invalid_argument( "invalid ngon: all points on line" );
+        }
+        if (k <= 3)
+        {
+            throw std::invalid_argument( "k must be 3 or higher" );
+        }
+        const size_t n = ngon.size();
+        if ((const int)n == k)
+        {
+            throw std::runtime_error ("(n = k)");
+        }
+        if ((const int)n < k)
+        {
+            throw std::runtime_error ("(n < k)");
+        }
+    }
+    catch (std::invalid_argument &message)
+    {
+        std::cout << "invalid argument: " << message.what() << std::endl;
+        return;
+    }
+    catch (std::runtime_error &message)
+    {
+        std::cout << "Warning: no minimum area polygon calculated " << message.what() << std::endl;
+        cv::Mat(ngon).copyTo(minPolygon);
+        area = cv::contourArea(minPolygon);
+        return;
+    }
+    catch (...)
+    {
+        std::cout << "input error" << std::endl;
+        return;
+    }
+
+    try
+    {
+        std::vector<cv::Point2f> kgon;
+        findMinAreaPolygon(ngon, kgon, area, k);
+        cv::Mat(kgon).copyTo(minPolygon);
+    }
+    catch (...)
+    {
+        std::cout << "correct input but execution failed" << std::endl;
+        return;
+    }
```

Please do not use try-catch. OpenCV uses CV_Assert, CV_Check for the function input validation and and CV_LOG_DEBUG/CV_LOG_INFO for notifications. Function should not silently ignore obvious invalid inputs like not enough points or wrong data types and ranges.